pkg/core/transaction/witness_condition_test.go

```diff
@@ -347,3 +352,52 @@ func TestWitnessConditionMatch(t *testing.T) {
 		require.Error(t, err)
 	})
 }
+
+func TestWitnessConditionCopy(t *testing.T) {
+	var someBool = true
+	boolCondition := (*ConditionBoolean)(&someBool)
+	pk, err := keys.NewPrivateKey()
+	require.NoError(t, err)
+
+	conditions := []WitnessCondition{
+		boolCondition,
+		&ConditionNot{Condition: boolCondition},
+		&ConditionAnd{boolCondition, boolCondition},
+		&ConditionOr{boolCondition, boolCondition},
+		&ConditionScriptHash{1, 2, 3},
+		(*ConditionGroup)(pk.PublicKey()),
+		ConditionCalledByEntry{},
+		&ConditionCalledByContract{1, 2, 3},
+		(*ConditionCalledByGroup)(pk.PublicKey()),
+		&ConditionNot{Condition: &ConditionNot{Condition: &ConditionNot{Condition: boolCondition}}},
+	}
+	for _, cond := range conditions {
+		copied := cond.Copy()
+		require.Equal(t, cond, copied)
+
+		switch c := copied.(type) {
+		case *ConditionBoolean:
+			require.NotSame(t, c, cond.(*ConditionBoolean))
+		case *ConditionScriptHash:
+			c[0]++
+			require.NotEqual(t, c, cond.(*ConditionScriptHash))
+		case *ConditionGroup:
+			c = (*ConditionGroup)(pk.PublicKey())
+			require.NotSame(t, c, cond.(*ConditionGroup))
+		case *ConditionCalledByContract:
+			c[0]++
+			require.NotEqual(t, c, cond.(*ConditionCalledByContract))
+		case *ConditionCalledByGroup:
+			c = (*ConditionCalledByGroup)(pk.PublicKey())
+			require.NotSame(t, c, cond.(*ConditionCalledByGroup))
+		case *ConditionNot:
+			require.NotSame(t, copied, cond)
+		case *ConditionAnd:
+			require.NotSame(t, copied, cond)
+		case *ConditionOr:
+			require.NotSame(t, copied, cond)
```

Comment:
For the nested we need to check not only this but also that if you change the child, then copy is not changed.